# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContextBuffers.cpp
# Commit: 7e2a2b1b416f
# Full Hash: 7e2a2b1b416f1a64c6548aab29e92acc9554ae7d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix all webgl regression tests according to CI. r=handyman
#   
#   (This is a combination of 31 commits)
#   
#   * Fix Linux compilation.
# ==============================================================================

diff -r ccfa767dba64 -r 7e2a2b1b416f dom/canvas/WebGLContextBuffers.cpp
--- a/dom/canvas/WebGLContextBuffers.cpp	Wed Jan 08 22:19:16 2020 +0000
+++ b/dom/canvas/WebGLContextBuffers.cpp	Wed Jan 08 22:19:23 2020 +0000
@@ -100,6 +100,11 @@
   const char* maxIndexEnum;
   switch (target) {
     case LOCAL_GL_TRANSFORM_FEEDBACK_BUFFER:
+      if (mBoundTransformFeedback->mIsActive &&
+          !mBoundTransformFeedback->mIsPaused) {
+        ErrorInvalidOperation("Transform feedback active and not paused.");
+        return nullptr;
+      }
       bindings = &(mBoundTransformFeedback->mIndexedBindings);
       maxIndexEnum = "MAX_TRANSFORM_FEEDBACK_SEPARATE_ATTRIBS";
       break;